src/App.vue

```diff
@@ -1,12 +1,41 @@
 <template>
   <div id="app">
-    <h1>hello world</h1>
+    <tx-preloader v-if="preload" :progress="progress" />
+
+    <main v-else>
+      <h1>hello world</h1>
+    </main>
   </div>
 </template>
 
 <script>
-export default {}
+import Preloader from "./assets/components/atoms/Preloader.vue";
+
+export default {
+  components: {
+    "tx-preloader": Preloader,
+  },
+  data() {
+    return {
+      preload: true,
+      progress: 0,
+    };
+  },
+  mounted() {
+    const preloadInterval = setInterval(() => {
+      if (this.progress < 100) {
+        this.progress++;
+      } else {
+        clearInterval(preloadInterval);
```

Comment:
is this working?

Reply:
yes, it is working fine.
does this code feel weird? or is there any other method that you know to do this correctly?

Reply:
i'll back to this later

Reply:
this is worked well. please continue